### PR TITLE
[HYDRATOR-1289] Changed implementation for Text type of format to use streaming

### PR DESCRIPTION
**Pull request by shashankmoghe**
https://issues.cask.co/browse/HYDRATOR-1289

### CODE REVIEW

File: src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java [hunk 1]
```diff
@@ -123,14 +122,13 @@ public void run(ActionContext context) throws Exception {
...
-                BufferedReader bufferedReader = new BufferedReader(reader);
-                String line;
-                while ((line = bufferedReader.readLine()) != null) {
-                  inputStringBuilder.append(line + "\n");
+              try (BufferedReader bufferedReader = new BufferedReader
+                (new InputStreamReader(inputStream, config.charset));) {
+                int i = 0;
+                char charsIn[] = new char[BUFFER_SIZE];
+                while ((i = bufferedReader.read(charsIn)) >= 0) {
+                  outputStream.write(new String(charsIn).getBytes(), 0, i);
```
Comment (russorat) [inline src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java:130]: Why convert it to String and then back to byte[]?

Comment (shashankmoghe) [reply src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java:130]: Other way to do convert char to byte would be as below.Please suggest
    byte[] bytes = new byte[charsIn.length];
    for (int i = 0; i < charsIn.length; i++) {
      bytes[i] = (byte) charsIn[i];
    }

Comment (russorat) [reply src/main/java/co/cask/hydrator/plugin/batch/HTTPToHDFSAction.java:130]: sorry, misread what you were doing.

**Comment by russorat** [comment]
LGTM

**Pull request merged**